src/org.ts

```diff
@@ -987,6 +1006,19 @@ export class Org extends AsyncCreatable<Org.Options> {
     });
   }
 
+  /**
+   *
+   * @param sandboxNameIn
+   * @returns sandboxInfoId
+   */
+  private async querySandboxInfoIdBySandboxName(sandboxNameIn: string): Promise<string> {
+    this.logger.debug('QuerySandboxInfoIdBySandboxName called with SandboxName: %s ', sandboxNameIn);
+    const queryStr = `SELECT Id, SandboxName FROM SandboxInfo WHERE SandboxName='${sandboxNameIn}'`;
+    const record = await this.connection.singleRecordQuery<{ Id: string }>(queryStr);
+    this.logger.debug('Return from calling queryToolingApi: %s ', record);
+    return record.Id;
```

Comment:
Is it possible that `record` is undefined from the query?  If so, this will throw a TypeError.

Reply:
singleRecordQuery throws unless there is exactly 1 record.